File: Backend/routes/rideRoutes.js
const express = require('express');
const router = express.Router();
const rideController = require('../controllers/rideController');
const {body,query} = require("express-validator");
const authMiddleware = require('../middleware/authMiddleware');


router.post('/create',
  authMiddleware.authUser,
  body('pickup').isString().isLength({min: 3}).withMessage('Invalid pickup location'),
  body('destination').isString().isLength({min: 3}).withMessage('Invalid destination location'),
  body('vehicleType').isIn(['auto', 'car', 'moto']).withMessage('Invalid vehicle type'),
  rideController.createRide
)
router.get('/get-fare',
    authMiddleware.authUser,
    query('pickup').isString().isLength({ min: 3 }).withMessage('Invalid pickup address'),
    query('destination').isString().isLength({ min: 3 }).withMessage('Invalid destination address'),
    rideController.getFare
)

router.post('/confirm',
    authMiddleware.authCaptain,
    body('rideId').isMongoId().withMessage('Invalid ride id'),
    rideController.confirmRide
)

router.get('/start-ride',
    authMiddleware.authCaptain,
    query('rideId').isMongoId().withMessage('Invalid ride id'),
    query('otp').isString().isLength({ min: 6, max: 6 }).withMessage('Invalid OTP'),
    rideController.startRide
)

router.post('/end-ride',
    authMiddleware.authCaptain,
    body('rideId').isMongoId().withMessage('Invalid ride id'),
    rideController.endRide
)


module.exports = router;

File: GoBackend/internal/routes/ride_routes.go
package routes

import (
	"uber/internal/controllers"
	"uber/internal/middleware"

	"github.com/gin-gonic/gin"
)

func RegisterRideRoutes(rg *gin.RouterGroup) {

	rides := rg.Group("/ride")
	{
		rides.GET("/get-fare", middleware.AuthMiddleware("users"), controllers.GetFare)
		rides.POST("/create", middleware.AuthMiddleware("users"), controllers.CreateRide)
		rides.POST("/pay", middleware.AuthMiddleware("users"), controllers.PayRide)

		rides.POST("/confirm", middleware.AuthMiddleware("captains"), controllers.ConfirmRide)

		rides.GET("/start", middleware.AuthMiddleware("captains"), controllers.StartRide)

		rides.POST("/end", middleware.AuthMiddleware("captains"), controllers.EndRide)
	}

	legacy := rg.Group("/rides")
	{
		legacy.GET("/get-fare", middleware.AuthMiddleware("users"), controllers.GetFare)
		legacy.POST("/create", middleware.AuthMiddleware("users"), controllers.CreateRide)
		legacy.POST("/pay", middleware.AuthMiddleware("users"), controllers.PayRide)
		legacy.POST("/confirm", middleware.AuthMiddleware("captains"), controllers.ConfirmRide)
		legacy.GET("/start-ride", middleware.AuthMiddleware("captains"), controllers.StartRide)
		legacy.POST("/end-ride", middleware.AuthMiddleware("captains"), controllers.EndRide)
	}
}
